### PR TITLE
BLO-780 fix: remove appstate for managing errors in lock screen

### PR DESCRIPTION
**Pull request by bluecco**
### Issue / feature description

`Tx error is displayed on login screen`

### Changes

- refactored lock screen and form to use local state instead of app state (had registered the tx error)
- tested locally with ionel too (to replicate, send max currency to another account and make another transaction while the first one is still pending)

### Checklist

- [x] Rebased to the last commit of the target branch (or merged)
- [x] Code self-reviewed
- [x] Code self-tested
- [x] Tests updated (if needed)
- [x] All tests are passing locally

Please keep your pull request as small as possible. If you need to make multiple changes, please create separate pull requests for each change. Create a draft pull request if you need early feedback. This will mark the pull request as a work in progress and prevent it from being reviewed or merged until you are ready.

Please move drafts to the ready for review (regular PR) when you are ready to start the review process and other developers will pick it up from there.

### CODE REVIEW

File: packages/extension/src/ui/features/lock/PasswordForm.tsx [hunk 1]
```diff
@@ -25,11 +26,9 @@ export const PasswordForm: FC<PasswordFormProps> = ({
   const { errors, isDirty, isSubmitting } = formState
   const { InfoIcon } = icons
 
-  const { error } = useAppState() // FIXME: as a hack we need to use global storage here, as the password form unmounts for the loading screen
```
Comment (diegodelrieu) [inline packages/extension/src/ui/features/lock/PasswordForm.tsx:28]: reading these comments I'm thinking there could be some side effects here, is it all good ?

Comment (bluecco) [reply packages/extension/src/ui/features/lock/PasswordForm.tsx:28]: yep, it was an old behaviour

basically it used the appState for loading (so the form was replaced by it)
we changed that in november to use a local loading (now the spinner is in the confirm button) but forgot to remove this part

File: packages/extension/src/ui/features/lock/LockScreen.tsx [hunk 2]
```diff
@@ -63,9 +64,7 @@ export const LockScreen: FC = () => {
                 navigate(target)
                 return true
               } catch {
-                useAppState.setState({
-                  error: "Incorrect password",
-                })
+                setError("Invalid password")
```
Comment (simonheys) [inline packages/extension/src/ui/features/lock/LockScreen.tsx:67]: Should keep the same wording - 'invalid' implies the password contains unexpected characters
```suggestion
                setError("Incorrect password")
```

File: packages/extension/src/ui/features/lock/LockScreen.tsx [hunk 3]
```diff
@@ -19,6 +18,7 @@ export const LockScreen: FC = () => {
   const navigate = useNavigate()
   const actions = useActions()
   const [isLoading, setIsLoading] = useState<boolean>(false)
+  const [error, setError] = useState<string | undefined>(undefined)
```
Comment (simonheys) [inline packages/extension/src/ui/features/lock/LockScreen.tsx:21]:
```suggestion
  const [error, setError] = useState<string>()
```

**Review by diegodelrieu — APPROVED** [review]

**Review by simonheys — APPROVED** [review]

**Pull request merged**